client/layout/index.jsx

```diff
@@ -200,14 +201,22 @@ class Layout extends Component {
 	}
 
 	shouldShowOdieAssistant() {
-		// We will only show the Odie Assistant under the "Upgrades" menu.
-		return (
+		// We will only show the Odie Assistant under the "Upgrades" menu. Every automattician should be able to see it under
+		// the feature flag, but we also want to show it to a percentage of users under the Wapuu Sales Agent experiment.
+
+		const isAutomattician =
 			[ 'plans', 'add-ons', 'domains', 'email', 'site-purchases', 'checkout' ].includes(
 				this.props.sectionName
 			) &&
 			! this.props.isOffline &&
-			config.isEnabled( 'odie' )
-		);
+			config.isEnabled( 'odie' );
+
+		const isTreatment =
+			this.props.wapuuSalesAgentExperimentAssignment?.variationName === 'treatment' &&
+			[ 'plans' ].includes( this.props.sectionName ) &&
+			! this.props.isOffline;
```

Comment:
You can see how the `isOffline` part gets repetitive (it's used in both conditions) and this is a signal that the logic could be improved/simplified.